primap2/pm2io/_data_reading.py

```diff
@@ -1244,17 +1310,150 @@ def rename_columns(
     return attrs
 
 
+_special_codes = {
+    "C": np.nan,
+    "NaN": np.nan,
+    "nan": np.nan,
+    "-": 0,
+    "NE0": np.nan,
+    "": np.nan,
+}
+
+
+def is_float(to_test: Any) -> bool:
+    try:
+        float(to_test)
+        return True
+    except ValueError:
+        return False
+
+
+def find_str_values_in_data(
+    data: pd.DataFrame,
+    columns: List[str],
+) -> list:
+    """Find all string values occurring in given columns of a DataFrame"""
+    # limit our analysis to columns that contain strings
+    # (or other object types)
+    cols_with_strs = (
+        data[columns].select_dtypes(include=[object]).columns.values.tolist()
+    )
+    temp = []
+    for col in cols_with_strs:
+        temp += list(data[col].unique())
+    temp = list(set(temp))
+    strs = [x for x in temp if not is_float(x)]
```

Comment:
`is_float("nan") == True` and therefore the string "nan" (and "NaN" and possible others) will never be included in the output of `find_str_values_in_data`. Does that cause problems? At least it is then unnecessary to include `"nan"` in `_special_codes`.

Reply:
I wondered why nan and NaN  were in the original list but didn't really think about it more and kept them. As `replace_values` does an explicit type conversion we don't need to replace them anyway as they will be converted fine. The explicit type conversion is necessary because without it we might get columns with dtype `int`. So I'll remove them from the list.

Reply:
Actually they need to be in the list, else parse code fails.